source/libs/sync/src/syncMain.c

```diff
@@ -51,15 +51,17 @@ static int32_t syncNodeAppendNoop(SSyncNode* ths);
 int32_t syncNodeOnPingCb(SSyncNode* ths, SyncPing* pMsg);
 int32_t syncNodeOnPingReplyCb(SSyncNode* ths, SyncPingReply* pMsg);
 
-// life cycle
-static void syncFreeNode(void* param);
 // ---------------------------------
+static void syncNodeFreeCb(void *param) {
+    syncNodeClose(param);
+    param = NULL;
+}
 
 int32_t syncInit() {
   int32_t ret = 0;
 
   if (!syncEnvIsStart()) {
-    tsNodeRefId = taosOpenRef(200, syncFreeNode);
+    tsNodeRefId = taosOpenRef(200, syncNodeFreeCb);
```

Comment:
Clean up all the resources (esp. those of constituent objects) allocated for the SSyncNode struct, based on refcount.